docs/WALKTHROUGH.md

```diff
@@ -0,0 +1,331 @@
+# Service Catalog Demonstration Walkthrough
+
+This document outlines the basic features of the service catalog by walking
+through a basic use case.
+
+## Step 1 - Installing the Service Catalog System
+
+The service catalog is conveniently packaged as a [helm](http://helm.sh/) 
+chart for installation.
+
+The chart is located in the [charts/catalog](../charts/catalog) directory in this repository, and 
+supports a wide variety of customizations which are laid out in the README.md in that
+directory. To install the service-catalog with sensible defaults, execute this command:
+
+```console
+helm install charts/catalog --name catalog --namespace catalog
+```
+
+## Step 2 - Understand Service Catalog Components
+
+Now that the system has been deployed to our Kubernetes cluster, we can use
+`kubectl` to talk to the service catalog API server.  The service catalog API
+has four resources:
+
+- Broker Server: A server that acts as a service broker and conforms to the 
+    [Open Service Broker API](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md)
+- `Broker` Resource: a representation of the broker server in the service-catalog system
+    that indicates where a broker server is running
+- `ServiceClass`: a service offered by a particular service broker. `ServiceClass`es are created
+    in response to a `Broker` resource being submitted to the system
+- `Instance`: an instance of a `ServiceClass` provisioned by the `Broker` for that `ServiceClass`
+- `Binding`: a binding to an `Instance` which is manifested into a Kubernetes namespace
+
+These components are building blocks of the service catalog in Kubernetes from an
+API standpoint.
+
+----
+
+#### Note: accessing the service catalog
+
+Unfortunately, `kubectl` doesn't know how to speak to both the service catalog
+API server and the main Kubernetes API server without switching contexts or
+`kubeconfig` files.  For now, the best way to access the service catalog API
+server is via a dedicated `kubeconfig` file.  You can manage the kubeconfig in
+use within a directory using the `direnv` tool.
+
+----
```

Comment:
As discussed, we should probably add something here about how to set up the service catalog api server. I will admit I don't really know what I'm doing or if there's a better way to do all this so please feel free to ignore or change the following suggestion:

> If you want to use multiple `kubeconfig` files, you need to specify set the `KUBECONFIG` environment variable. For example, you can run
> 
> ```
> KUBECONFIG=<path_to_config> kubectl <command>
> ```
> 
> To make it less tedious, you can set up an alias, so for example, if you wanted to set your config file for the service catalog API server in your current directory you would run
> 
> ```
> alias catalogctl="KUBECONFIG=$(pwd)/kubeconfig kubectl"
> ```
> 
> To set up the service catalog api server in kubeconfig, run
> 
> ```
> catalogctl config set-credentials service-catalog-creds --username=admin --password=admin
> catalogctl config set-cluster service-catalog-cluster --server=http://${D_HOST}:${PORT}
> catalogctl config set-context service-catalog-ctx --cluster=service-catalog-cluster --user=service-catalog-creds
> catalogctl config use-context service-catalog-ctx
> ```
> 
> If you aren't using the alias then replace `catalogctl` with `KUBECONFIG=<path_to_config> kubectl`
> 
> Now you can use the `catalogctl` command whenever you want to talk to the service catalog API server.

Of course.. the issue with this is that you will have to change the rest of the walkthrough to use catalogctl instead..